src/manipulators/no_implicit_any_manipulator.ts

```diff
@@ -26,9 +37,286 @@ export class NoImplicitAnyManipulator extends Manipulator {
+        this.errorDetector.filterDiagnosticsByCode(
+          diagnostics,
+          this.errorCodesToFix
+        )
+      ),
+      this.nodeKinds
+    );
+
+    // Set of declarations with implicit any type.
+    const modifiedDeclarations = new Set<AcceptedDeclaration>();
+
+    // Iterate through each node, adding declarations to set.
+    errorNodes.forEach(({node: errorNode}) => {
+      if (Node.isIdentifier(errorNode)) {
+        const errorSymbol = errorNode.getSymbol();
+        const declarations = errorSymbol?.getDeclarations();
+        declarations?.forEach(declaration => {
+          if (
+            Node.isVariableDeclaration(declaration) ||
+            Node.isParameterDeclaration(declaration)
+          ) {
+            modifiedDeclarations.add(declaration);
+          }
+        });
+      }
+    });
+
+    // Map from declaration to calculated type of declaration.
+    const determinedTypes = new Map<AcceptedDeclaration, Set<Type>>();
+
+    // Graph of dependencies between declarations.
+    const dependencyGraph = new Map<
+      AcceptedDeclaration,
+      Set<AcceptedDeclaration>
+    >();
+
+    modifiedDeclarations.forEach(declaration => {
+      // For parameter declarations, go though each call expression of the parent function.
+      if (Node.isParameterDeclaration(declaration)) {
+        // Get the parent function declaration
+        const parentFunctionDeclaration = declaration.getFirstAncestorByKind(
+          SyntaxKind.FunctionDeclaration
+        );
+
+        // Get index of parameter declaration
+        const parameterIndex = parentFunctionDeclaration
+          ?.getParameters()
+          ?.indexOf(declaration);
+
+        // Search for function call references
+        const functionReferences = parentFunctionDeclaration?.findReferencesAsNodes();
+        functionReferences?.forEach(functionReference => {
+          // In call reference, get arguments in those calls
+          const args = functionReference
+            .getParentIfKind(SyntaxKind.CallExpression)
+            ?.getArguments();
+
+          // Add corresponding argument's declaration to the dependency graph
+          if (
+            args &&
+            parameterIndex !== undefined &&
+            args.length > parameterIndex
+          ) {
+            const correspondingArg = args[parameterIndex];
+            this.addDependency(
+              dependencyGraph,
+              determinedTypes,
+              modifiedDeclarations,
+              correspondingArg,
+              declaration
+            );
+          }
+        });
+      }
+
+      // For all declarations, find assignment references.
+      const references = declaration.findReferencesAsNodes();
+      references?.forEach(reference => {
+        const parent = reference.getParentIfKind(SyntaxKind.BinaryExpression);
+        const sibling = reference.getNextSiblingIfKind(SyntaxKind.EqualsToken);
+        const nextSibling = sibling?.getNextSibling();
```

Comment:
let's use more meaningful variable name.

parent/sibling/nextSibling may be meaningful from AST perspective, can we use more straight forward language like 
parent -> binaryExpressionNode
sibling-> equlaTokenNode
nextSibling-> assignmentValueNode

etc

so is others.

Reply:
renamed variables